### PR TITLE
Ft sign up user api 157581730

### PR DESCRIPTION
**Pull request by LuffyKing**
#### What does this PR do?
This fixes Travis Postgres database issues, incorporates LFA feedback, adds a signup API for users to register on the app with,  and robust tests for said API.

#### Description of Task to be completed?
Fixing CI database and env variables configuration to ensure the testing framework is in lockstep with the app. Addition of robust tests to ensure edge cases are accounted for and handled properly.  Signup API created to allow users to register on the app and take advantage of the routes that require a logged in user.
#### Any background context you want to provide?
Robust testing is the key to a solid application.
#### What are the relevant Github Issues?
The only issue so is only the pending review
Questions:

### CODE REVIEW

File: server/db/indexTest.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
+DROP TABLE IF EXISTS USERS,REQUESTS;
+CREATE TABLE IF NOT EXISTS USERS(
+ID UUID PRIMARY KEY NOT NULL,
+FIRST_NAME VARCHAR(80) NOT NULL,
+LAST_NAME VARCHAR(80) NOT NULL,
+EMAIL VARCHAR(80) NOT NULL UNIQUE,
+PASSWORD VARCHAR(90) NOT NULL,
```
Comment (Tawakalt) [inline server/db/indexTest.sql]: Don't you think `90` character spaces is too much for PASSWORD? Same with others.
